bob-backend/src/pearl/group.rs

```diff
@@ -170,54 +172,37 @@ impl Group {
     // create pearl for current write
     async fn create_write_pearl(
         &self,
-        ts: u64,
+        data: &BobData,
         timestamp_config: StartTimestampConfig,
     ) -> Result<(ChildId, Holder), Error> {
-        let mut indexes = self.created_holder_indexes.write().await;
-        let created_holder_index = indexes.get(&ts).copied();
-        let index = if let Some(exisiting_index) = created_holder_index {
-            exisiting_index
+        let holders = self.holders.upgradable_read().await;
```

Comment:
Please add a comment that only one thread can hold the `upgradable_read` lock at a time. This is crucial for proper working